src/NuGet.Core/NuGet.PackageManagement/Resolution/ResolverGather.cs

```diff
@@ -214,7 +221,17 @@ public int MaxDegreeOfParallelism
                     throw new InvalidOperationException(message);
                 }
             }
-
+            // calculate total time taken to gather all packages as well as with each soource
+            var type = string.Empty;
+            stopWatch.Stop();
+            var timeDiff = DatetimeUtility.ToReadableTimeFormat(stopWatch.Elapsed, out type);
+            _context.Log.LogMinimal(string.Format("Gather dependency information took {0:0.##} {1}", timeDiff, type));
+            _context.Log.LogDebug("Below is the summary of time taken to gather dependencies source-wise :");
```

Comment:
Summary of time take to gather dependencies per source

Indent the individual results to a tab after this for easier readability